### PR TITLE
Add SnakeLines tool

### PR DESCRIPTION
**Pull request by wernerkrampl**
Added recipe for SnakeLines tool

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by wernerkrampl** [comment]
@BiocondaBot please add label

**Review by dpryan79 — DISMISSED** [review]
There should really be a wrapper script that's knows where the snakefiles are relative to it, since users rarely actually know where files are installed by conda packages (this is good, it shouldn't matter). Thus, it would be quite difficult for them to know what the path to snakelines.snake is. Adding wrapper programs in python that end up calling `snakemake` internally is a convenient way around this.

**Review by dpryan79 — CHANGES_REQUESTED** [review]
Once the package is installed users will have no clue where snakelines.snake ends up getting installed (sure, they can figure it out, but they're unlikely to want to). You might ask the authors to add wrapper scripts that call `snakemake` internally with appropriate settings to make this more useful.

File: recipes/snakelines/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{% set name = "snakelines" %}
+{% set version = "0.10.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/jbudis/snakelines/archive/master.zip
+  sha256: bb323936dd4a73b8e34b31126e9319feca364166979e8ea4020cc36d0fc1146e
```
[the remark this answers is not in the document]
Comment (wernerkrampl) [reply recipes/snakelines/meta.yaml:10]: There was a miscommunication in our team. Fixed by setting the correct release number in meta.yaml to 0.9.2.

File: recipes/snakelines/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - pandas ==1.1.3 
+    - biopython ==1.78
+    - seaborn ==0.11.0
+    - bs4 ==4.9.3
+    - weasyprint ==51
+    - pysam ==0.16.0.1
+    - openpyxl ==3.0.5
+    - scikit-bio ==0.5.6
+    - jinja2 ==2.11.2
+    - snakemake ==5.13.0
```
[the remark this answers is not in the document]
Comment (wernerkrampl) [reply recipes/snakelines/meta.yaml:30]: Lot of the packages share their dependencies with each other, but there are conflicts and this is currently the only way we were able to let it run. However, we will be updating this in the near future.

File: recipes/snakelines/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+test:
+  commands:
+    - snakemake -v
```
[the remark this answers is not in the document]
Comment (wernerkrampl) [reply recipes/snakelines/meta.yaml]: Fixed by adding a wrapper and testing `share/snakelines/snakelines.snake` in an included genomic example.

File: recipes/snakelines/build.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+mkdir -pv $PREFIX/bin/
+cp -rv docs enviroments example legacy pipeline rules src License.txt README.rst config.yaml snakelines.snake $PREFIX/bin/
```
[the remark this answers is not in the document]
Comment (wernerkrampl) [reply recipes/snakelines/build.sh]: Fixed by moving the snakelines files to `$PREFIX/share/snakelines` and by adding a `snakelines` wrapper to `$PREFIX/bin`.

**Comment by wernerkrampl** [comment]
I have added a wrapper script `snakelines` to `$PREFIX/bin` calling `$PREFIX/share/snakelines` as a parameter for `snakemake`. Now user just have to call `snakelines` and provide a configfile.

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**